### PR TITLE
Added Grouped List Item in the Demo App

### PR DESCRIPTION
**Pull request by Dhruv-Mishra**
### Problem 
Request For Grouped List Item

### Fix
Added the Demo for GroupedListItem in the DemoApp

### Screenshots
image

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] Automated Tests
- [ ] Documentation and demo app examples
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and RTL layouts
- [ ] Size classes and window sizes (notched devices, multitasking, different window sizes, etc)

### CODE REVIEW

File: FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/demos/V2ListItemActivity.kt [hunk 1]
```diff
@@ -275,10 +281,49 @@ private fun CreateListActivityUI(context: Context) {
...
+            text = "Text",
+            onClick = {},
+            textAlignment = ListItemTextAlignment.Regular,
+            border = BorderType.Top,
+            trailingAccessoryContent = { Icon(icon = FluentIcon(Icons.Outlined.KeyboardArrowRight))},
+        )
+    }
+    ListItem.SectionDescription(description = "Grouped list containing multiple similar elements", modifier = Modifier.wrapContentHeight().padding(0.dp))
+}
+
```
Comment (joyeeta26) [inline FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/demos/V2ListItemActivity.kt:326]: please add both before and after screenshots

Comment (Dhruv-Mishra) [reply FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/demos/V2ListItemActivity.kt:326]: 1. This did not exist before, so only added the after screenshot. 
2. The requirement request for a group list Item as per a provided Figma. I added it to the Demo App to dismiss the requirement for a separate component and show that ListItem covers the required functionality already.

File: FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/demos/V2ListItemActivity.kt [hunk 2]
```diff
@@ -275,10 +281,49 @@ private fun CreateListActivityUI(context: Context) {
...
 
+@Composable
+private fun GroupedList() {
+    ListItem.Header("Grouped List")
+    ListItem.SectionDescription(description = "Grouped List", modifier = Modifier.height(25.dp))
+    Column(modifier = Modifier.padding(horizontal = 10.dp, vertical = 10.dp).clip(
+        RoundedCornerShape(10.dp))) {
+        ListItem.Item(
+            text = "Text",
+            onClick = {},
```
Comment (joyeeta26) [inline FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/demos/V2ListItemActivity.kt:297]: can we use loop for this?

Comment (Dhruv-Mishra) [reply FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/demos/V2ListItemActivity.kt]: Updated

**Review by joyeeta26 — APPROVED** [review]

**Pull request merged**